### PR TITLE
chore(NODE-6269): Update dev dependencies and add dependabot config

### PR DESCRIPTION
**Pull request by baileympearson**
#### What is changing?

- Development dependencies are updated.  Notably, I adopted eslint v9 but filed a follow-up ticket to migrate our config files to the new flat configuration file format.
- A basic dependabot config has been added.  This config ignores any dependencies that can't be upgraded past their current major and runs monthly.

### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: test/node/bson_compliance_test.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ describe('BSON Compliance', function () {
         // Attempt to deserialize
         BSON.deserialize(buffer);
         expect(false).to.be.ok;
-      } catch (err) {
+      } catch {
```
Comment (baileympearson) [inline test/node/bson_compliance_test.js:24]: eslint now correctly reports `err` as unused in these tests.

File: .eslintrc.json [hunk 2]
```diff
@@ -51,20 +51,16 @@
...
     ],
-    "@typescript-eslint/ban-types": [
-      "error",
-      {
-        "types": {
-          "Function": false
-        }
-      }
-    ],
+    "@typescript-eslint/no-unsafe-function-type": "off",
```
Comment (baileympearson) [inline .eslintrc.json:54]: Superceeded by `no-unsafe-function`, which is enabled by default from the typescript-eslint recommended rules package.

File: .eslintrc.json [hunk 3]
```diff
@@ -51,20 +51,16 @@
...
     "@typescript-eslint/no-unsafe-member-access": "off",
     "@typescript-eslint/no-unsafe-assignment": "off",
     "@typescript-eslint/no-unsafe-return": "off",
     "@typescript-eslint/no-unsafe-argument": "off",
     "@typescript-eslint/no-unsafe-call": "off",
     "@typescript-eslint/no-unsafe-enum-comparison": "off",
+    // Note: you must disable the base rule as it can report incorrect errors
+    "no-unused-expressions": "off",
+    "@typescript-eslint/no-unused-expressions": "off",
```
Comment (baileympearson) [inline .eslintrc.json:63]: Literally hundreds of failures here, many of them in tests from mocha assertions (i.e., expect(...)... returns an assertion, so it is technically an unused expression).

We could just disable this in our test suite, but for consistency I disabled it everywhere.

Comment (aditi-khare-mongoDB) [reply .eslintrc.json:63]: If we've disabled 'no-unused-expressions,' is there a reason we need to remove the unused `error`s from the test files?

**Review by aditi-khare-mongoDB — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 3 above)
Comment (baileympearson) [reply .eslintrc.json:63]: Different lint rules 🙂 the `errors` are removed from no-unused-vars, not no-unused-expressions.

Comment (aditi-khare-mongoDB) [reply .eslintrc.json:63]: ahh i see!

**Review by aditi-khare-mongoDB — APPROVED** [review]
LGTM!

**Pull request merged**